site/source/tutorials/working_with_virtual_dom.md

```diff
@@ -0,0 +1,106 @@
+
+```ts
+  const pageTitle = document.createElement('h1');
+  pageTitle.innerHTML = 'Welcome to Biz-E-Corp!';
+  pageTitle.classList.add('title');
+  document.body.appendChild(pageTitle);
+```
+
+This example starts by creating the `h1` element and then setting two properties - the `innerHTML` property specifies the element's contents and the `classList.add()` method adds a CSS class to the element to alter its appearance. Finally, the element is added to the `document`'s `body` element via a call to that element's `appendChild` method.
+
+The DOM APIs are well-known and reliable, but it can be challenging to use them efficiently. This challenge is caused by the way in which web browsers coordinate changes in the DOM and the page. Whenever a DOM change changes the visible layout of the page, the browser must redetermine the location of each element and redraw them. This means that mundane changes to the DOM can significantly degrade an application's performance.
+
+## Advantages of virtualizing the DOM
+
+> "We can solve any problem by introducing an extra level of indirection" - David J. Wheeler
+
+Ensuring that the DOM is updated efficiently is a critical component to maximizing a web application's performance. Doing that consistently across a large application, however, can be very challenging. Rather than relying on developers to manage this manually, Dojo 2 uses an abstraction layer to manage DOM updates automatically. This abstraction layer is called a *virtual* DOM.
+
+A virtual DOM has several advantages when compared to traditional DOM manipulation. A virtual DOM improves developer productivity by allowing them to interact with a simple, consistent API in a declarative manner. It also improves performance by gathering DOM updates into batches and applying them at the optimum time. Finally, applications using a virtual DOM use resources more efficiently since applications interact with lightweight representations of DOM elements instead of the DOM elements themselves.
+
+The DOM API has existed for many years and has been continually expanded to keep up with the evolution of web browsers. This evolution has yielded a programming interface that is occasionally verbose and inconsistent. Virtual DOM libraries are able to reconsider the use-cases for managing page elements and expose a simpler set of interfaces for developers to work with. For example, consider the example in the preceeding section where we created an `h1` tag, set some properties on it, and add it to the page. In Dojo 2, this would be accomplished using the following code:
+
+```ts
+  v('h1', { class: 'title' }, [ 'Welcome to Biz-E-Corp' ]);
+```
+
+A single line of code is all that is required to accomplish what the DOM API required four to do. By providing a simple, declarative API, rendering code is typically much cleaner and easier to understand.
```

Comment:
After first sentence, consider adding "While lines of code is not the only way to measure productivity, frequently used APIs greatly benefit engineering productivity when they are efficient and consistent.